torchx/schedulers/test/ray_scheduler_test.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import Any, Iterator, Type, cast
+from unittest import TestCase, skipIf
+from unittest.mock import patch
+
+from torchx.schedulers import get_schedulers
+
+try:
+    from torchx.schedulers.ray_scheduler import RayScheduler, logger
+
+    should_skip = False
+except ImportError:
+    should_skip = True
+
+from torchx.specs import AppDef, Resource, Role, RunConfig, runopts
+
+
+@skipIf(should_skip, "Ray is not installed.")
```

Comment:
it might be better to wrap these in If statements -- internally it gets annoying since skip tests show up as action items

same way we handle docker https://github.com/pytorch/torchx/blob/main/torchx/schedulers/test/docker_scheduler_test.py#L102